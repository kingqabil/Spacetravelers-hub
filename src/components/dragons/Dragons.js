import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Button from 'react-bootstrap/Button';
import { getDragonsFromAPI } from '../../redux/dragons/dragons';

const Dragons = () => {
  const dispatch = useDispatch();
  const dragonsData = useSelector((state) => state.dragons);
  const getDragons = () => {
    if (dragonsData.length === 0) {
      dispatch(getDragonsFromAPI());
    }
  };
  useEffect(() => {
    getDragons();
  }, []);

  return (
    <div className="dragons-list">
      {
dragonsData.map((dragon) => (
  <div key={dragon.id} className="dragon">
    <div className="dragon-img">
      <img src={dragon.dragon_img} alt={dragon.dragon_name} />
    </div>
    <div className="dragon-description">
      <h3>{dragon.dragon_name}</h3>
      <p>{dragon.dragon_description}</p>
      <Button variant="primary">Reserve Dragon</Button>
    </div>
  </div>
))
}
    </div>
  );
};

export default Dragons;
