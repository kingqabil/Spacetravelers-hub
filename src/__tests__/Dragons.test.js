import React from 'react';
import { Provider } from 'react-redux';
import {render} from '@testing-library/react'
import Dragons from '../components/dragons/Dragons';
import store from '../redux/configureStore';

const ReactTestRenderer = require('react-test-renderer');

describe('Dragon component tests', () => {

	it('renders correctly', () => {
		const tree = ReactTestRenderer
			.create(
				<Provider store={store}><Dragons /></Provider>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
  
	it('has a text in the DOM', () => {
		const { container } = render(<Provider store={store}><Dragons /></Provider>)
		expect(container.getElementsByClassName('dragons-list').length).toBe(1);
})
})

