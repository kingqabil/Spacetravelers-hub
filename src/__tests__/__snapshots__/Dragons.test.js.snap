// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon component tests renders correctly 1`] = `
<div
  className="dragons-list"
/>
`;
